diagnostics/tropical_pacific_sea_level/tropical_pacific_sea_level.py

```diff
@@ -0,0 +1,671 @@
+# - dynamical_balance2.curl_tau    : calculate wind stress curl in model (for Dataset without time dim)
+# - xr_ufunc.da_linregress : linregress for Dataset with time dim
+#
+
+
+import os
+import cftime
+# import dask
+import xarray as xr
+import numpy as np
+import matplotlib.pyplot as plt
+
+import spherical_area as sa
+from xr_ufunc import da_linregress
+from dynamical_balance2 import curl_tau, curl_tau_3d
+from dynamical_balance2 import curl_var, curl_var_3d
+
+# from dask.distributed import Client
+# client = Client(n_workers=1, threads_per_worker=8, processes=False)
+# client
+
+
+import warnings
+warnings.simplefilter("ignore")
+
+######
+# testing parameter before combining with MDTF framework
+#####
+
+
+# #### possible input info from external text file
+# # constant setting
+# syear = 1993                 # crop model and obs data from year
+# fyear = 2009                 # crop model and obs data to year
+# tp_lat_region = [-30,30]     # extract model till latitude
+
+# # regional average box
+# lon_range_list = [120,150]   # 0-360
+# lat_range_list = [10,20]
+
+# # Model label
+# Model_name = ['CESM2_CORE']        # model name in the dictionary
+# Model_legend_name = ['CESM2-CORE'] # model name appeared on the plot legend
+
+# # initialization
+# modelin = {}
+# path = {}
+# #####################
+# ori_syear = 1948
+# ori_fyear = 2009
+# modeldir = os.getenv('MODEL_DATA_ROOT')+'/CESM2_omip1_r1i1p1f1_gn/mon/'
+# modelfile = [['CESM2_omip1_r1i1p1f1_gn.tauuo.mon.nc'],
+#              ['CESM2_omip1_r1i1p1f1_gn.tauvo.mon.nc'],
+#              ['CESM2_omip1_r1i1p1f1_gn.zos.mon.nc']]
+# areafile = 'CESM2_omip1_r1i1p1f1_gn.areacello.mon.nc'
+# path[Model_name[0]]=[modeldir,modelfile]
+
+# Model_varname = ['tauuo','tauvo','zos']
+# Model_dimname = ['time','nlat','nlon']
+# Model_coordname = ['lat','lon']
+
+# xname = Model_dimname[2]
+# yname = Model_dimname[1]
+
+print('--------------------------')
+print('Start reading set parameter (pod_env_vars)')
+print('--------------------------')
+# constant setting
+syear = np.int(os.getenv('syear'))                 # crop model and obs data from year
+fyear = np.int(os.getenv('fyear'))                  # crop model and obs data to year
+tp_lat_region = [-30,30]                 # extract model till latitude
+
+# regional average box
+lon_range_list = [np.float(os.getenv('lon_min')),
+                  np.float(os.getenv('lon_max'))]   # 0-360
+lat_range_list = [np.float(os.getenv('lat_min')),
+                  np.float(os.getenv('lat_max'))]
+
+# Model label
+Model_name = [os.getenv('Model_name')]        # model name in the dictionary
+Model_legend_name = [os.getenv('Model_legend_name')] # model name appeared on the plot legend
+
+
+
+# initialization
+modelin = {}
+path = {}
+#####################
+ori_syear = int(os.getenv('FIRSTYR'))
+ori_fyear = int(os.getenv('LASTYR'))
+modeldir = str(os.getenv('DATADIR'))+"/../"+str(os.getenv('Model_path'))
```

Comment:
These definitions should be removed: absolute paths to all requested variables are set via the `path_variable` setting in the new settings file format, which defaults to `<variable name>_FILE`; see [documentation](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/ref_settings.html#varlist-entry-properties). Calls to open_dataset(), etc. should also be changed.